libgadget/fof.c

```diff
@@ -801,8 +802,9 @@ static void fof_reduce_groups(
     /* count how many we have of each task */
     memset(Send_count, 0, sizeof(int) * NTask);
 
-    for(i = 0; i < nmemb; i++) {
-        struct BaseGroup * gi = (struct BaseGroup *) (((char*) groups) + i * elsize);
+    size_t n;
```

Comment:
It feels a bit strange to iterate with an unsigned variable.

Reply:
I recall reading otherwise. This is an interesting read:

https://software.intel.com/en-us/forums/intel-c-compiler/topic/534769

C++ does use unsigned indices in the standard library, but scroll to the 'we were young' part. 

I think size_t is for specifying sizes, and ptrdiff_t is for offsets (indexing, which can be signed). Some even argue size_t shall be signed as well. But anyhow, I can deal with it if indices are unsigned unless necessary.

Reply:
The STL does it was indeed my reasoning - I think the important thing is that the loop index not be a smaller type than the limit variable. A cleaner fix might just be to make Ngroups etc signed.

Reply:
Can we rely on Ngroups being < 2^31? I think we probably can.